crates/agglayer/src/cli.rs

```diff
@@ -52,4 +52,71 @@ pub(crate) enum Commands {
     },
 
     Vkey,
+
+    #[clap(subcommand)]
+    Backup(Backup),
+}
+
+#[derive(Subcommand)]
+pub(crate) enum Backup {
+    /// List all backups.
+    List {
+        #[arg(long, short, value_hint = ValueHint::FilePath, default_value = "agglayer.toml", env = "CONFIG_PATH")]
+        config_path: PathBuf,
+    },
+
+    /// Restore from a backup.
+    Restore {
+        #[arg(long, short, value_hint = ValueHint::FilePath, default_value = "agglayer.toml", env = "CONFIG_PATH")]
+        config_path: PathBuf,
+        #[arg(value_parser = parse_db_kind_version)]
+        db_versions: Vec<(DbKind, u32)>,
+    },
+}
+
+#[derive(Debug, Clone)]
+pub(crate) enum DbKind {
+    State,
+    Pending,
+    Epoch(u64),
+}
+
+impl std::str::FromStr for DbKind {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str().trim() {
+            "state" => Ok(DbKind::State),
+            "pending" => Ok(DbKind::Pending),
+            s if s.starts_with("epoch_") => {
+                if let Some(s) = s.strip_prefix("epoch_") {
```

Comment:
Nit: The `strip_prefix` and `starts_with` perform redundant checks.
```suggestion
            s => {
                if let Some(s) = s.strip_prefix("epoch_") {
```

(May need to adjust the error messages and the wildcard branch can be removed after this.)